ndb/src/google/cloud/ndb/_eventloop.py

```diff
@@ -259,6 +263,64 @@ def run(self):
                 break
 
 
+class _LocalContexts(threading.local):
+    """Maintain a thread local stack of event loops."""
+
+    def __init__(self):
+        self.stack = []
+
+    def push(self, loop):
+        self.stack.append(loop)
+
+    def pop(self):
+        return self.stack.pop(-1)
+
+    def current(self):
+        if self.stack:
+            return self.stack[-1]
+
+
+contexts = _LocalContexts()
+
+
+@contextlib.contextmanager
+def async_context():
+    """Establish an asynchronous context for a set of asynchronous API calls.
+
+    This function provides a context manager which establishes the event loop
+    that will be used for any asynchronous NDB calls that occur in the context.
+    For example:
+
+    .. code-block:: python
+
+        from google.cloud.ndb import async_context
+
+        with async_context():
+            # Make some asynchronous calls
+            pass
+
+    Within the context, any calls to a ``*_async`` function or to an
+    ``ndb.tasklet``, will be added to the event loop established by the
+    context.  Upon exiting the context, execution will block until all
```

Comment:
I've been avoiding the "typewriter" double space after a sentence. (H/T to @tseaver for "typewriter" reference.)

Reply:
I've been trying to stop as well, but it's a 20+ year habit.